docs/index.bs

```diff
@@ -2970,16 +2972,34 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
               1. For each |eventType| of |settingsObject|'s [=environment settings object/global object=]'s associated list of <a>event listeners</a>' event types:
                   1. [=set/Append=] |eventType| to |workerGlobalScope|'s associated [=ServiceWorkerGlobalScope/service worker=]'s <a>set of event types to handle</a>.
 
-                  Note: If the global object's associated list of event listeners does not have any event listener added at this moment, the service worker's set of event types to handle remains an empty set. The user agents are encouraged to show a warning that the event listeners must be added on the very first evaluation of the worker script.
+                  Note: If the global object's associated list of event listeners does not have any event listener added at this moment, the service worker's set of event types to handle remains an empty set.
 
               1. Set |script|'s <a>has ever been evaluated flag</a>.
+              1. If the [=No-Op Handler identification=] algorithm with |serviceWorker| returns true, the [=has had non-no-op fetch listeners flag=] is set to false.
```

Comment:
This seems backwards. I don't think we ever want to unset this flag. Or have I misunderstood the design?


```suggestion
              1. If [=No-Op Handler identification=] given |serviceWorker| is false, then set |serviceWorker|'s [=has had non-no-op fetch listeners flag=].
```

When referring to a property like `[=has had non-no-op fetch listeners flag=]`, be clear about which object you're referring to.

Reply:
Let me update the flag name.  The current flag name requires to have the default value because we should not think fetch handlers are no-op by default.